synapse/handlers/federation.py

```diff
@@ -353,15 +362,11 @@ async def _maybe_backfill_inner(
         # First we try hosts that are already in the room
         # TODO: HEURISTIC ALERT.
 
-        curr_state = await self.state_handler.get_current_state(room_id)
-
-        curr_domains = get_domains_from_state(curr_state)
-
-        likely_domains = [
-            domain for domain, depth in curr_domains if domain != self.server_name
-        ]
+        users_in_room = await self.store.get_users_in_room(room_id)
+        likely_domains = {get_domain_from_id(u) for u in users_in_room}
```

Comment:
Actually, `get_domains_from_state` was doing more here and ordering the servers by oldest user first.